clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -634,12 +634,12 @@ public boolean matches(ClientRequest request) {
 
 
     @Test
-    public void testExcludeInternalTopicsConfigOption() { 
+    public void testExcludeInternalTopicsConfigOption() {
         subscriptions.subscribe(Pattern.compile(".*"), rebalanceListener);
 
         metadata.update(TestUtils.singletonCluster(TopicConstants.GROUP_METADATA_TOPIC_NAME, 2), time.milliseconds());
 
-        assertFalse(subscriptions.partitionAssignmentNeeded());
+        assertTrue(subscriptions.partitionAssignmentNeeded());
```

Comment:
One scenario I can think of is when a topic that matches the regex is altered (e.g. its number of partitions changes). Currently, this does not seem to trigger a rebalance; even though a rebalance is required. So, it sounds like a rebalance should be done in regex subscription to cover cases like this. I hope that makes sense.